internal/entities/properties/properties.go

```diff
@@ -202,12 +202,32 @@ type Properties struct {
 	props *xsync.MapOf[string, Property]
 }
 
-// NewProperties Properties from a map
-func NewProperties(props map[string]any) (*Properties, error) {
+// newPropertiesOption is a function that configures NewProperties
+type newPropertiesOption func(*newPropertiesConfig)
+
+type newPropertiesConfig struct {
+	skipPrefixCheck bool
+}
+
+// WithSkipPrefixCheckTestOnly returns an option to skip checking the prefix
+// This should only be used for testing purposes
+func WithSkipPrefixCheckTestOnly() newPropertiesOption {
```

Comment:
If this is for testing purposes only, let's move it to the test file itself. It should be fine as it's part of the same package.